engine/src/lib.rs

```diff
@@ -69,20 +62,66 @@ impl ServalEngine {
     /// Run the passed-in WASM executable on the given input bytes.
     pub fn execute(
         &mut self,
+        // WebAssembly binary to execute
         binary: &[u8],
+        // Data to pass to WebAssembly as stdin
         input: &[u8],
+        // List of elevated permissions for this binary
+        permissions: &[Permission],
     ) -> Result<WasmResult, ServalEngineError> {
         let stdout = WritePipe::new_in_memory();
         let stderr = WritePipe::new_in_memory();
 
+        // Link the experimental HTTP support
+        let allowed_http_hosts = if permissions.contains(&Permission::AllHttpHosts) {
+            // todo: unclear whether we should actually support a wildcard like this
+            vec!["insecure:allow-all".to_string()]
+        } else {
+            permissions
+                .iter()
+                .filter_map(|perm| match perm {
+                    Permission::HttpHost(origin) => Some(origin.to_owned()),
+                    _ => None,
+                })
+                .collect()
+        };
+
+        if !allowed_http_hosts.is_empty() {
+            let http_state =
+                HttpState::new().map_err(ServalEngineError::EngineInitializationError)?;
+            http_state
+                .add_to_linker(&mut self.linker, move |_| HttpCtx {
+                    // todo: there's some confusion around whether
+                    allowed_hosts: Some(allowed_http_hosts.clone()),
+                    max_concurrent_requests: Some(42),
```

Comment:
_Can_ a single Wasm module actually make 42 different concurrent requests, considering it single-threaded?

Reply:
Yeah, because the actual work is being done over in Rust Land.